index.js

```diff
@@ -27,8 +27,12 @@ export class Minibus {
 
   /** @param {import('multiformats').CID} cid */
   async has (cid) {
-    // https://github.com/web3-storage/minibus/issues/7
-    return Boolean(await this.get(cid))
+    const url = new URL(base58btc.encode(cid.multihash.bytes), this.endpoint)
+    const res = await this.fetch(url, {
+      method: 'HEAD',
+      headers: this.headers
+    })
+    return res.ok
```

Comment:
We shouldn't smother the error here - non-404/200 is unexpected and should be thrown:

```suggestion
    if (res.status === 404) return false
    if (!res.ok) {
      throw new Error(`failed to determine existence of ${cid}: server responded with status: ${res.status}`)
    }
    return true
```